tests/multi_agents/test_supervisor_multi_agent.py

```diff
@@ -0,0 +1,99 @@
+from unittest.mock import AsyncMock, MagicMock
+
+import pytest
+from langchain_core.language_models import GenericFakeChatModel
+from langchain_core.messages import HumanMessage, SystemMessage
+from neuroagent.multi_agents.supervisor_multi_agent import AgentState
+
+from src.neuroagent.multi_agents import SupervisorMultiAgent
+
+
+def test_create_main_agent():
+    mock_llm = MagicMock()
+    bind_function_result = MagicMock()
+    bind_function_result.__ror__.return_value = {}
+    mock_llm.bind_functions.return_value = bind_function_result
+    data = {"llm": mock_llm, "agents": [("agent1",)]}
+    from src.neuroagent.multi_agents import SupervisorMultiAgent
+
+    result = SupervisorMultiAgent.create_main_agent(data)
+    assert "main_agent" in result
+    assert "summarizer" in result
+
```

Comment:
this is rather test_create_main_agent_initialization. 
i was going to suggest to add assert agent1 in agents but i see it in test_create_graph.